go-controller/pkg/ovn/egressgw.go

```diff
@@ -391,32 +391,28 @@ func (oc *Controller) deletePodGWRoutesForNamespace(pod *kapi.Pod, namespace str
 						continue
 					}
 					mask := GetIPFullMask(podIP)
-					node := util.GetWorkerFromGatewayRouter(gr)
-					portPrefix, err := oc.extSwitchPrefix(node)
-					if err != nil {
-						klog.Infof("Failed to find ext switch prefix for %s %v", node, err)
-						continue
-					}
-
 					if err := oc.deleteLogicalRouterStaticRoute(podIP, mask, gwIP.String(), gr); err != nil {
 						klog.Errorf("Unable to delete pod %s route to GR %s, GW: %s, err:%v",
 							pod, gr, gwIP.String(), err)
 						klog.Error(err)
 					} else {
+						delete(routeInfo.podExternalRoutes[podIP], gwIP.String())
 						klog.V(5).Infof("ECMP route deleted for pod: %s, on gr: %s, to gw: %s", pod,
 							gr, gwIP.String())
+					}
 
-						delete(routeInfo.podExternalRoutes[podIP], gwIP.String())
-						// clean up if there are no more routes for this podIP
-						if entry := routeInfo.podExternalRoutes[podIP]; len(entry) == 0 {
-							// TODO (trozet): use the go bindings here and batch commands
-							// delete the ovn_cluster_router policy if the pod has no more exgws to revert back to normal
-							// default gw behavior
-							if err := oc.delHybridRoutePolicyForPod(net.ParseIP(podIP), node); err != nil {
-								klog.Error(err)
-							}
+					node := util.GetWorkerFromGatewayRouter(gr)
+					if entry := routeInfo.podExternalRoutes[podIP]; len(entry) == 0 {
+						if err := oc.delHybridRoutePolicyForPod(net.ParseIP(podIP), node); err != nil {
+							klog.Error(err)
 						}
 					}
+
+					portPrefix, err := oc.extSwitchPrefix(node)
```

Comment:
before if we couldn't get the port prefix we would skip processing this entire iteration. Now we will remove the ecmp route and hybrid policy. This is a problem because we remove the key from the podExternalRoutes map, so next time if we try to delete this thing we will think we dont need to, but the bfd entry was never cleaned up. right?

Reply:
Per slack, fixed by ensuring that if cleaning the BFD entry fails, we don't remove the route from the pod's list. Try again later.

Reply:
I dont see it fixed here in this first commit, maybe you fixed it later, but it should be fixed here.